### PR TITLE
Refactor pose_estimator package

### PR DESCRIPTION
**Pull request by hidmic**
# Description

This patch refactors the `pose_estimator` package to:

- Publish `auv_interfaces::StateStamped` messages
- Drop `pose_estimator::CorrectedData` message
- Normalize style and layout

Connected to #87.

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [ ] Unit tests are passing
- [x] Linter tests are passing

# How To Test

## Test A

Launch pose estimation subsystem.

```sh
roslaunch system_bringup pose_estimation.launch
```

Use `rostopic echo` to make sure it is publishing as expected.

### CODE REVIEW

**Comment by hidmic** [comment]
Rebased on top of #93.

File: pose_estimator/launch/pose_estimator.launch [hunk 1]
```diff
@@ -2,18 +2,20 @@
...
+	<arg name="maxYawAng" default="6.283185307"/>
 	
 	<!-- pose_estimator -->
 	<node pkg="pose_estimator" type="pose_estimator_node" name="pose_estimator_node" output="screen">
 		<param name="rate" type="double" value="100.0"/>
 		<param name="min_rate" type="double" value="50.0"/>
 		<param name="max_rate" type="double" value="200.0"/>
-		<param name="saltwater_flag" type="bool" value="false"/>
+		<param name="in_saltwater" type="bool" value="false"/>
+    <param name="rpm_per_knot" type="double" value="303.0"/>
```
Comment (patapufate) [inline pose_estimator/launch/pose_estimator.launch:13]: nit

**Review by patapufate — APPROVED** [review]
LGTM

File: pose_estimator/launch/pose_estimator.launch (hunk 1 above)
Comment (hidmic) [reply pose_estimator/launch/pose_estimator.launch:13]: Yeah, DOS carriage return keeps coming back.

**Comment by hidmic** [comment]
Thanks for the review @patapufate. This needs an update though. It is now publishing faster than its inputs, which causes an stagnation error. Have to synchronize output.

**Comment by hidmic** [comment]
Now it's up-to-date @patapufate. PTAL!

**Review by patapufate — APPROVED** [review]
LGTM

**Comment by hidmic** [comment]
Thank ! Going in.

**Pull request merged**